src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitString.Fixer.cs

```diff
@@ -0,0 +1,111 @@
+                        createChangedDocument: async c =>
+                        {
+                            SemanticModel semanticModel = await doc.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                            var operation = semanticModel.GetOperationWalkingUpParentChain(expression, cancellationToken);
+                            if (operation is IVariableDeclaratorOperation variableDeclaratorOperation)
+                            {
+                                return await HandleVariableDeclarator(variableDeclaratorOperation, doc, root, cancellationToken).ConfigureAwait(false);
+                            }
+                            else if (operation is IArgumentOperation argumentOperation)
+                            {
+                                if (argumentOperation.Value is ILiteralOperation argumentLiteral)
+                                {
+                                    return await HandleStringLiteral(argumentLiteral, doc, root, cancellationToken).ConfigureAwait(false);
+                                }
+                            }
+                            return doc;
+                        },
+                        equivalenceKey: "PreferConstCharOverConstUnitStringInStringBuilderAppend"),
+                    context.Diagnostics);
+            }
+        }
+
+        private static async Task<Document?> HandleStringLiteral(ILiteralOperation argumentLiteral, Document doc, SyntaxNode root, CancellationToken cancellationToken)
+        {
+            if (argumentLiteral.ConstantValue.HasValue && argumentLiteral.ConstantValue.Value is string unitString && unitString.Length == 1)
+            {
+                DocumentEditor editor = await DocumentEditor.CreateAsync(doc, cancellationToken).ConfigureAwait(false);
+                SyntaxGenerator generator = editor.Generator;
+                char charValue = unitString[0];
+                SyntaxNode charLiteralExpressionNode = generator.LiteralExpression(charValue);
+                var newRoot = generator.ReplaceNode(root, argumentLiteral.Syntax, charLiteralExpressionNode);
+                return doc.WithSyntaxRoot(newRoot);
+            }
+            return doc;
+        }
+
+        private static async Task<Document?> HandleVariableDeclarator(IVariableDeclaratorOperation variableDeclaratorOperation, Document doc, SyntaxNode root, CancellationToken cancellationToken)
+        {
+            IVariableDeclarationOperation variableDeclarationOperation = (IVariableDeclarationOperation)variableDeclaratorOperation.Parent;
+            if (variableDeclarationOperation == null)
+            {
+                return null;
+            }
+
+            IVariableDeclarationGroupOperation variableGroupDeclarationOperation = (IVariableDeclarationGroupOperation)variableDeclarationOperation.Parent;
+            if (variableGroupDeclarationOperation.Declarations.Length != 1)
```

Comment:
This is the part that detects if there is more than one variable declared in one line, correct? As in: `string a = "a", b = "b";`

Reply:
It's the next if block. Line 83. In Roslyn, a `VariableDeclaration` statement can have multiple `VariableDeclarators`. I'm not actually sure what it means to have `variableGroupDeclarationOperation.Declarations.Length > 1`, but I have it here just to be safe